src/core/RandomVariables.jl

```diff
@@ -186,7 +186,7 @@ function VarInfo(old_vi::UntypedVarInfo, spl, x::AbstractVector)
     return new_vi
 end
 function VarInfo(old_vi::TypedVarInfo, spl, x::AbstractVector)
-    md = newmetadata(old_vi.metadata, getspaceval(spl), x)
```

Comment:
This seems unrelated to the actual PR. Is this change necessary?

Reply:
Some of the changes in RandomVariables.jl are needed but I'll move the rest to a separate PR.

Reply:
Thanks!